css-view-transitions-2/Overview.bs

```diff
@@ -753,6 +753,66 @@ div.box {
 		1. Return |viewTransition|.
 	</div>
 
+# Determining 'view-transition-name' automatically # {#auto-vt-name}
+
+## Overview ## {#auto-vt-name-overview}
+
+*This section is non-normative.*
+
+For an element to participate in a view transition, it needs to receive a unique 'view-transition-name'.
+This can be tideous and verbose when multiple elements are involved in the same view transition, especially in cases where many of those are same-element transitions,
+as in, the element has the same 'view-transition-name' in the old and new state.
+
+To make this easier, setting the 'view-transition-name' to ''view-transition-name/auto'' would generate a 'view-transition-name' for the element, or take it from the element's [=Element/id=] if present.
+
+## Examples ## {#auto-vt-name-example}
+
+<div class="example">
+In this example, view transition is used to sort a list in an animated way:
+
+```html
+<ul>
+	<li>Item 1</li>
+	<li>Item 2</li>
+	<li>Item 3</li>
+	<li>Item 4</li>
+	...
+</ul>
+```
+
+Ordinarily, each of these items would have to receive a unique 'view-transition-name':
+
+```css
+li:nth-child(1) { view-transition-name: item1; }
+li:nth-child(2) { view-transition-name: item1; }
+li:nth-child(3) { view-transition-name: item1; }
+li:nth-child(4) { view-transition-name: item1; }
+...
+```
+
+With ''view-transition-name/auto'', this CSS would work:
+
+```css
+li {
+	view-transition-name: auto;
+}
+```
+</div>
+
+## Additions to 'view-transition-name' ## {#additions-to-vt-name}
+
+In addition to the existing values, the 'view-transition-name' also accepts an <dfn for=view-transition-name type=value export>auto</dfn> keyword.
+To resolve the [=used value=] of 'view-transition-name' for |element|:
+	1. Let |computed| be the [=computed value=] of 'view-transition-name'.
+	1. If |computed| is <css>none</css>, return null.
+	1. If |computed| is a <<custom-ident>>, return |computed|.
+	1. Assert: |computed| is <css>auto</css>.
+	1. If |element| has an associated [=Element/id=], then return |element|'s [=Element/id=].
+	1. Return a unique string, which is not a <<custom-ident>>. This string should remain consistent and unique for this element and {{Document}}, at least for the lifetime of |element|'s [=node document=]'s [=active view transition=].
+
+		Note: this string is not web-observable, and is used for addressing the element in internal algorithms.
```

Comment:
Maybe also leave a note that this means for cross-document view transitions, elements using a unique string will turn into exit animations.

Reply:
Will do. Actually I need to make a change that keeps the `ID` check tree-scoped, so will ask for a re-review.

Reply:
Actually do we need any scoping text here? It looks like we do that when querying the computed value for `view-transition-name` here: https://drafts.csswg.org/css-view-transitions-1/#document-scoped-view-transition-name.

So maybe just the change the step 1 here to say, "Let |computed| be the be the element’s document-scoped view transition name." linking back to that algo?

Note to self: A bunch of places like these will need to change for scoped transitions. :)

Reply:
Yes, because we read the ID. The ID might have a different tree-scope than the style where the `view-transition-name` property is defined (if it's a `::part`), we need to specifically account for that and only read the ID if it's in the same scope.

Reply:
Ah, the ID is a good point. Right now since there's only document scoped transitions, the "document-scoped view transition name" will be not none only if it was set via `::part`. In which case it has outer tree scope.

But the id attribute on the element should always be the inner tree scope, irrespective of how it was set. Correct me if I got it wrong (just ran it past @tabatkins). I'm assuming so because inline styles from the style attribute are also tied to the inner tree scope.

So does that imply `view-transition-name: auto` set via `::part` can never use the id based name..?

Reply:
Correct, look at the note above (row 812)

Reply:
Got it. In that case I'm having a hard time understanding the wording of the id point. 😅 How about something like this: "If |element| has an associated [=Element/id=], and the tree scope of |computed| is the same as the tree scope of the element, then ...". Because that's what the code does right? We compare the tree scope of the node and the computed value.

And I feel like we still need to mention that we only apply the computed value if it's tree scope matches the Document? It's getting a bit ambiguous on how this interacts with the algorithm in L1 that reads the names.

Reply:
Sure, that would have the same meaning. We do have the other tree scope wording in L1 already

Reply:
Done

Reply:
> Sure, that would have the same meaning. We do have the other tree scope wording in L1 already

Yea but the way it's written right now it's not clear if it's overriding what's defined in L1. For example why are we only looking at the tree scope for ID but not the custom ident? I'm wondering if we should just phrase this as an addition to the "document-scoped view transition name" algorithm. That already deals with how we go from computed value to used value.

Reply:
> > Sure, that would have the same meaning. We do have the other tree scope wording in L1 already
> 
> 
> 
> Yea but the way it's written right now it's not clear if it's overriding what's defined in L1. 

I don't understand. This is just for `auto` which is not in L1

Reply:
I just meant that the used value for `auto` happens after we do a tree scope check on the computed value (defined in L1) right? So should we say that somewhere here, the algorithm right now just says we're using the computed value directly. And returns it for the custom ident case.

Reply:
L1 just says `Each [=view transition name=] is a [=tree-scoped name=]`. We don't change this, I don't think we need to add anything. The only special thing we do is check that scope vs the element's ID's scope if needed

Reply:
Oh I see now, let me fix this

Reply:
Check now